gateway/session_manager.go

```diff
@@ -418,3 +442,33 @@ func GetAccessDefinitionByAPIIDOrSession(session *user.SessionState, api *APISpe
 
 	return accessDef, allowanceScope, nil
 }
+
+type endpointRateLimitInfo struct {
```

Comment:
use embedding for rate/per to avoid type conversions, move to internal/model and add any godoc comments as necessary please